### PR TITLE
Custom sandboxing implementation as linux usernamespace calls - port cachepot/#128

### PR DESCRIPTION
**Pull request by drahnr**
Implementes a custom sandboxing implementation as linux usernamespace calls.

It's opt-in, and has to be enabled with setting the `SCCACHE_SANDBOX` environment variable.

Ref #1620 

Original work done by @Xanewok

### CODE REVIEW

**Comment by drahnr** [comment]
@sylvestre `/sccache-dist: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory` not sure how to resolve this, do you have an insight into the container env?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -48,6 +48,15 @@ jobs:
             rustc: 1.60.0
             extra_desc: dist-server
             extra_args: --no-default-features --features=dist-tests test_dist_ -- --test-threads 1
+            sccache_log: sccache=trace
+            dist_exec_strategy: docker
+          - os: ubuntu-20.04
```
Comment (Xuanwo) [inline .github/workflows/ci.yml:53]: PLease use ubuntu-22.04 for openssl-3.0

Comment (drahnr) [inline .github/workflows/ci.yml:53]:
```suggestion
          - os: ubuntu-22.04
```

**Comment by sylvestre** [comment]
dunno much about this one but I guess it is working with cachepot, why not :)

**Comment by drahnr** [comment]
It does work, I'll add some documentation once the safety concern in in the comments is addressed - it should be considered experimental.

**Pull request merged**